cli/wallet/file/file.go

```diff
@@ -0,0 +1,348 @@
+package file
+
+import (
+	"crypto/rand"
+	"crypto/sha512"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	"github.com/mitchellh/mapstructure"
+	flag "github.com/spf13/pflag"
+	bip39 "github.com/tyler-smith/go-bip39"
+	"golang.org/x/crypto/pbkdf2"
+
+	"github.com/oasisprotocol/deoxysii"
+	"github.com/oasisprotocol/oasis-core/go/common/crypto/sakg"
+	coreSignature "github.com/oasisprotocol/oasis-core/go/common/crypto/signature"
+
+	"github.com/oasisprotocol/oasis-sdk/cli/config"
+	"github.com/oasisprotocol/oasis-sdk/cli/wallet"
+	"github.com/oasisprotocol/oasis-sdk/client-sdk/go/crypto/signature"
+	"github.com/oasisprotocol/oasis-sdk/client-sdk/go/crypto/signature/ed25519"
+	"github.com/oasisprotocol/oasis-sdk/client-sdk/go/types"
+)
+
+const (
+	walletKind = "file"
+
+	cfgAlgorithm = "algorithm"
+	cfgNumber    = "number"
+
+	algorithmEd25519    = "ed25519"
```

Comment:
seeing as the Algorithm field covers the derivation algorithm and the resulting key type, maybe this should be ed25519-adr0008?

Reply:
or no, this is used in both walletConfig and secretState, which kind of conflates the meaning.

when walletConfig.Algorithm is algorithmEd25519, it means "loading this wallet will give you an ed25519 key." but it's stored next to a Number field, which is rather something that controls a derivation not hinted at by this field

when secretState.Algorithm is algorithmEd25519, it means "use ADR 0008 to derive an ed25519 key from the mnemonic." it's stored next to that mnemonic.

some weirdness from all this:

- the choice between ADR 0008 derivation and raw private key is within the sealed data, but the account number for the ADR 0008 case is out in the open
- there's this piece of code that switches around algorithmEd25519Raw based on a field in ImportSource

Reply:
Yeah that's ugly, I've switched this over so that there's now `ed25519-adr8` and `ed25519-raw` in the open. The number field is indeed something that is specific to ADR8 derivation though.